integration/cdi/src/main/java/org/apache/cxf/cdi/SingletonResourceProvider.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.cxf.cdi;
+
+import org.apache.cxf.jaxrs.lifecycle.ResourceProvider;
+import org.apache.cxf.message.Message;
+
+public class SingletonResourceProvider implements ResourceProvider {
+    private final Class<?> clazz;
+    private Object instance;
+
+    SingletonResourceProvider(final Lifecycle lifecycle, final Class<?> clazz) {
+        this.instance = lifecycle.create();
```

Comment:
Should we create instance proactively here? It is simpler like that but it may not be needed (in theory, at least)

Reply:
Well for all normal scoped beans it is still lazy thanks to CDI proxies and for @Singleton/@Dependent it becomes eager but it is ok from a business perspective + the extension already does it in several cases so not a big deal IMHO.